frontend/src/app/components/header.js

```diff
@@ -12,6 +12,15 @@ import { withRouter } from 'react-router-dom'
 const logo = require('./../../media/logo.png')
 
 class HeaderComponent extends Component {
+  changeTheme = () => {
+    this.props.changeTheme()
+
+    // Stop transition effects
+    window.setTimeout(() => {
```

Comment:
This should ideally go to a saga so that the React component itself doesn't handle side effects. You'll probably want something like a saga that gets fired on every `CHANGE_THEME`, waits for the timeout period and then dispatches the `STOP_THEME_TRANSITION`. This also allows you to completely omit the `stopThemeTransition()` from `Header` props.

Reply:
you also don't need to write _window_ explicitly

Reply:
Done and done

Reply:
Resolved [here](https://github.com/salsita/foosball-rating/pull/81/commits/25ba48df7bd72b6d383bdf8f576df28eb479576d#diff-bf7b3488ddec0115f2387242335ffae0)

> you also don't need to write _window_ explicitly

done [here](https://github.com/salsita/foosball-rating/pull/81/commits/25ba48df7bd72b6d383bdf8f576df28eb479576d#diff-a5b8f7d4dfd7d1c2c65b70a166a77b8f) and [here](https://github.com/salsita/foosball-rating/pull/81/commits/25ba48df7bd72b6d383bdf8f576df28eb479576d#diff-bf7b3488ddec0115f2387242335ffae0) , although I still use window during acessing localStorage. Shall I remove this as well @novellizator?